### ISSUE TITLE
How to use the compressed model in tensorflow

### ISSUE THREAD

**Issue by jimmystereo** [quic/aimet#786]
The problem is that the accuracy of the compressed model is not same as the one identified in stats.
So I wonder that if I have misused it.
Here's the code and outputs:
### Input 1
```"""
Convert an AIMET  spatial SVD compressed session to a Keras model and train the Keras model with MirroredStrategy
"""
sess = get_sess_from_keras_model()
conv2d = sess.graph.get_operation_by_name('conv2d/Conv2D')
modules_to_ignore = [conv2d]

greedy_params = GreedySelectionParameters(target_comp_ratio=Decimal(0.8),
                                          num_comp_ratio_candidates=10,
                                          use_monotonic_fit=True,
                                          saved_eval_scores_dict=None)

auto_params = SpatialSvdParameters.AutoModeParams(greedy_select_params=greedy_params,
                                                      modules_to_ignore=modules_to_ignore)

# Create random dataset
batch_size = 1
input_data = np.random.rand(100, 32, 32, 3)
dataset = tf.data.Dataset.from_tensor_slices(input_data)
dataset = dataset.batch(batch_size=batch_size)


params = SpatialSvdParameters(input_op_names=['input_1'],
                                  output_op_names=['dense/Softmax'],
                                  mode=SpatialSvdParameters.Mode.auto,
                                  params=auto_params,
                                  multiplicity=8)

# Single call to compress the model
results = ModelCompressor.compress_model(sess,
                                         working_dir=None,
                                         eval_callback=evaluate_model,
                                         eval_iterations=10,
                                         input_shape=(32, 32, 32, 3),
                                         compress_scheme=CompressionScheme.spatial_svd,
                                         cost_metric=CostMetric.mac,
                                         parameters=params)

compressed_model, stats = results
compressed_ops = ['conv2d/Conv2D']

input_op_name, output_op_name = "input_1:0", "dense/Softmax:0"

# Step 1: Single Saving the compressed session
path = './saved_model_single_gpu'
save_tf_session_single_gpu(compressed_model, path, input_op_name, output_op_name)
acc = evaluate_model(sess, 1, 1)
acc2 = evaluate_model(compressed_model, 1, 1)

tf.keras.backend.clear_session()

# Step 2: Loading the correspnding Keras Model
tf.keras.backend.set_learning_phase(1)
loaded = load_tf_sess_variables_to_keras_single_gpu(path, compressed_ops)
sess.close()
```
### Input 2
```
print(acc)
print(acc2)
print(stats)
```
### output 2
```
0.8897
0.1016
**********************************************************************************************
Compressed Model Statistics
Baseline model accuracy: 0.889700, Compressed model accuracy: 0.796600
Compression ratio for memory=0.685626, mac=0.790312
```

### input 3
```
loaded.compile(
    optimizer=optimizer, loss='categorical_crossentropy', metrics=['accuracy'])
loss_accuracy_train = loaded.evaluate(train_data)
print("Training Loss: {:.4f}".format(loss_accuracy_train[0]))
print("Training Accuracy: {:.2%}".format(loss_accuracy_train[1]))
loss_accuracy = loaded.evaluate(valid_data)
print("Validation Loss: {:.4f}".format(loss_accuracy[0]))
print("Validation Accuracy: {:.2%}".format(loss_accuracy[1]))
```

### output 3
```
1/1 [==============================] - 32s 32s/step - loss: 2.9064 - acc: 0.1015
Training Loss: 2.9064
Training Accuracy: 10.15%
1/1 [==============================] - 6s 6s/step - loss: 2.8765 - acc: 0.1016
Validation Loss: 2.8765
Validation Accuracy: 10.16%
```

As shown from the stats, the correct accuracy of the output model should be 0.796600.
But I can't reproduce the result from each way. Is there any mistake in the usage?
Thanks!

**Comment by quic-ssiddego** [comment]
@jimmystereo  Thank you for reporting this. Tagging @quic-hitameht  and @quic-sundarr to take a look at this and suggest. (specifically - comment on spatial svd support for _keras_ model)

**Comment by quic-hitameht** [comment]
@quic-ssiddego Yes we do support spatial SVD for keras model.
@jimmystereo We use returned 'compressed_model' session only to calculate compressed model accuracy. Can you share your 'evaluate_model' function to understand this better?

**Comment by quic-ssiddego** [comment]
@jimmystereo Just checking back - do you have further inputs/ queries on this?

**Comment by quic-mangal** [comment]
Closing this issue due to inactivity. Please re-open it/ create a new issue if you need further help.
